### DISCUSSION TITLE
distinct view

### DISCUSSION THREAD

**Discussion by nick9one1** [rappen/FetchXMLBuilder#1062] [Q&A]
I'm new here and have a fairly basic question I hope someone can answer. 

I want to generate FetchXML to retrieve a list of distinct site names from a property table. 

This is pretty much what I want;

![image](https://github.com/user-attachments/assets/82a71622-66cd-43f0-8531-4faec908c990)

but if I try to save this as a view I get an error -  
![image](https://github.com/user-attachments/assets/202781fa-4805-4358-8be7-8b271d75f5da)

If I add this attribute (below) the table is no longer distinct. The result view goes from 18 results to 500.
I presume the distinct filter is now looking at the propertytree4id field.

`<fetch top="50" distinct="true">
  <entity name="cr20b_propertytree4">
    <attribute name="cr20b_sitecode" />
    <attribute name="cr20b_sitename" />
    <attribute name="cr20b_sitesyscode" />
    <attribute name="cr20b_propertytree4id" />
    <order attribute="cr20b_sitename" />
  </entity>
</fetch>`

How can I fix this? thanks!

**Comment by rappen** [comment]
I assume you want to have this query in a view in Dynamics, and that's why you try to save the view, correct?
A view should have the ID of each record so that you can open the record on each row in the result. This is why FXB complains about it when saving a view.
It should be possible to just ignore this by answering "No". It will make the view "not as it should", but I guess it may work anyway...

I don't understand why, though. I guess there is another table that just contains the sites, right?
